akeneo-design-system/src/components/Input/TagInput/TagInput.tsx

```diff
@@ -0,0 +1,223 @@
+   */
+  allowDuplicates: boolean;
+
+  /**
+   * Tags to display
+   */
+  tags: string[];
+
+  /**
+   * Handle called when tags are updated
+   */
+  setTags: (tags: string[]) => void;
+
+  /**
+   * Placeholder displayed where there is no tag
+   */
+  placeholder?: string;
+
+  /**
+   * Defines if the input is valid on not.
+   */
+  isInvalid?: boolean;
+};
+
+const MAX_TAGS = 100;
+
+const TagInput: FC<TagInputProps> = ({allowDuplicates, setTags, placeholder, isInvalid, tags = []}) => {
+  const [isLastTagSelected, selectLastTag] = useState<boolean>(false);
+  const inputRef = useRef<HTMLInputElement>(null);
+  const containerRef = useRef<HTMLUListElement>(null);
+  const inputContainerRef = useRef<HTMLLIElement>(null);
+
+  const updateTags = useCallback(
+    (updatedTags: string[]) => {
+      updatedTags = updatedTags.slice(0, Math.min(MAX_TAGS, updatedTags.length));
+      setTags(updatedTags);
+    },
+    [setTags]
+  );
+
+  const onChangeCreateTags = useCallback(
+    (event: ChangeEvent<HTMLInputElement>) => {
+      const tagsAsString = event.currentTarget.value;
+      if (tagsAsString !== '') {
+        let newTags = tagsAsString.split(/[\s,;]+/);
+        if (newTags.length === 1) {
+          return;
+        }
+        newTags = newTags.filter((tag: string) => tag.trim() !== '');
+
+        createTags([...tags, ...newTags]);
+      }
+    },
+    [tags]
+  );
+
+  const onBlurCreateTag = useCallback(
+    (event: ChangeEvent<HTMLInputElement>) => {
+      const inputCurrentValue = event.currentTarget.value.trim();
+      if (inputCurrentValue !== '') {
+        createTags([...tags, ...[inputCurrentValue]]);
+      }
+    },
+    [tags]
+  );
+
+  const createTags = useCallback(
+    (newTags: string[]) => {
+      if (!allowDuplicates) {
+        newTags = arrayUnique(newTags);
+      }
+      updateTags(newTags);
+      if (inputRef && inputRef.current) {
+        inputRef.current.value = '';
+      }
+    },
+    [arrayUnique, inputRef, updateTags]
+  );
+
+  const removeTag = useCallback(
+    (tagIdToRemove: number) => {
+      const clonedTags = [...tags];
+      clonedTags.splice(tagIdToRemove, 1);
+      updateTags(clonedTags);
+    },
+    [tags, updateTags]
+  );
+
+  const focusOnInputField = useCallback(
+    (event: MouseEvent, ref: RefObject<HTMLElement>) => {
+      if (ref && ref.current === event.target && inputRef && inputRef.current) {
+        inputRef.current.focus();
+      }
+    },
+    [inputRef]
+  );
+
+  const handleTagDeletionUsingKeyboard = useCallback(
+    (event: KeyboardEvent) => {
+      if (![Key.Backspace.toString(), Key.Delete.toString()].includes(event.key)) {
+        selectLastTag(false);
+
+        if (tags.length >= MAX_TAGS) {
+          event.preventDefault();
+          event.stopPropagation();
+        }
+
+        return;
+      }
+
+      if (tags.length === 0 || (inputRef && inputRef.current && inputRef.current.value.trim() !== '')) {
```

Comment:
```suggestion
      const fieldisEmpty = inputRef && inputRef.current && inputRef.current.value.trim() !== ''
      const tagsAreEmpty = tags.length === 0
      if (fieldisEmpty || tagsAreEmpty) {
```

I find this kind of naming way clearer